tests/contrib/mariadb/test_mariadb.py

```diff
@@ -0,0 +1,278 @@
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.service == "mariadb"
+    assert span.name == "mariadb.connection.rollback"
+
+
+def test_analytics_default(connection, tracer):
+    cursor = connection.cursor()
+    cursor.execute("SELECT 1")
+    rows = cursor.fetchall()
+    assert len(rows) == 1
+    spans = tracer.pop()
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.get_metric(ANALYTICS_SAMPLE_RATE_KEY) is None
+
+
+@pytest.mark.parametrize(
+    "service_env_key,service_env_value", [("DD_SERVICE", "mysvc"), ("DD_MARIADB_SERVICE", "mysvc")]
+)
+def test_user_specified_service_snapshot(run_python_code_in_subprocess, service_env_key, service_env_value):
+    """
+    When a user specifies a service for the app
+        The mariadb integration should not use it.
+    """
+
+    @snapshot(
+        async_mode=False,
+        token_override="tests.contrib.mariadb.test_mariadb.test_user_specified_service_snapshot_{}_{}".format(
+            service_env_key, service_env_value
+        ),
+    )
+    def testcase():
+        out, err, status, pid = run_python_code_in_subprocess(
+            """
+from ddtrace import config
+from ddtrace import patch
+import mariadb
+patch(mariadb=True)
+from tests.contrib.config import MARIADB_CONFIG
+connection = mariadb.connect(**MARIADB_CONFIG)
+cursor = connection.cursor()
+cursor.execute("SELECT 1")
+rows = cursor.fetchall()
+assert len(rows) == 1
+""",
+            env={service_env_key: service_env_value},
+        )
+        assert status == 0, err
+
+    testcase()
+
+
+@snapshot(include_tracer=True)
+def test_simple_query_snapshot(tracer):
+    connection = get_connection(tracer)
+    cursor = connection.cursor()
+    cursor.execute("SELECT 1")
+    rows = cursor.fetchall()
+    assert len(rows) == 1
+
+
+@snapshot(include_tracer=True, ignores=["meta.error.stack"])
+def test_simple_malformed_query_snapshot(tracer):
+    connection = get_connection(tracer)
+    cursor = connection.cursor()
+    with pytest.raises(mariadb.ProgrammingError):
+        cursor.execute("SELEC 1")
+
+
+@snapshot(include_tracer=True)
+def test_simple_query_fetchall_snapshot(tracer):
+    with override_config("mariadb", dict(trace_fetch_methods=True)):
+        connection = get_connection(tracer)
+        cursor = connection.cursor()
+        cursor.execute("SELECT 1")
+        rows = cursor.fetchall()
+        assert len(rows) == 1
+
+
+@snapshot(include_tracer=True)
+def test_query_with_several_rows_snapshot(tracer):
+    connection = get_connection(tracer)
+    cursor = connection.cursor()
+    query = "SELECT n FROM (SELECT 42 n UNION SELECT 421 UNION SELECT 4210) m"
+    cursor.execute(query)
+    rows = cursor.fetchall()
+    assert len(rows) == 3
+
+
+@snapshot(include_tracer=True)
+def test_query_with_several_rows_fetchall_snapshot(tracer):
+    with override_config("mariadb", dict(trace_fetch_methods=True)):
+        connection = get_connection(tracer)
```

Comment:
do these connections need a `with connection: pass`? are they going to get cleaned up after each test case?